File: leetcode_150/array/majority_element_169.py
from typing import List
from collections import Counter

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        target = dict(Counter(nums))
        size = len(nums) / 2
        for n in target:
            if target[n] >= size:
                return n


ss = Solution()

ss.majorityElement([2,2,1,1,1,2,2])

File: leetcode_150/array/majority_element_169.java
class Solution {
    public int majorityElement(int[] nums) {
        var holder = 0;
        var candidate = 0;
        for (int n : nums) {
            if (holder == 0) {
                candidate = n;
            }
            holder += (n == candidate ? 1 : -1);
        }
        return candidate;
    }
}